app/src/main/java/net/squanchy/tweets/view/ScrollListener.java

```diff
@@ -0,0 +1,47 @@
+package net.squanchy.tweets.view;
+
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+public abstract class ScrollListener extends RecyclerView.OnScrollListener {
+
+    private static final int THRESHOLD = 2;
+
+    private int previousTotalItemCount = 0;
+    private boolean loading = true;
+
+    private final LinearLayoutManager layoutManager;
+
+    protected ScrollListener(LinearLayoutManager layoutManager) {
+        this.layoutManager = layoutManager;
+    }
+
+    @Override
+    public void onScrolled(RecyclerView view, int dx, int dy) {
+        int totalItemCount = layoutManager.getItemCount();
```

Comment:
do you need the layout manager as a dependency? You receive the recycler view as a parameter